src/integrationTest/java/uk/gov/hmcts/darts/workflow/command/AddAudioMidTierCommand.java

```diff
@@ -45,36 +59,102 @@ public AddAudioMidTierCommand(String ipaddress, Audio audio, File fileToSend) {
         exitCode = 0;
     }
 
+    public AddAudioMidTierCommand(String ipaddress, File audioFile, File fileToSend) {
+        this.ipToGateway = ipaddress;
+        this.audioFile = audioFile;
+        this.fileToSend = fileToSend;
+        exitCode = 0;
+    }
+
     @Override
     public void cleanupResources() {
         if (tempFile != null && tempFile.exists()) {
             tempFile.delete();
         }
     }
 
+    private File getAudioFile() throws IOException {
+        if (audio != null) {
+            try {
+                tempFile = File.createTempFile("audio", ".xml");
+                JAXBContext jaxbContext = JAXBContext.newInstance(Audio.class);
+                Marshaller marshaller = jaxbContext.createMarshaller();
+                marshaller.marshal(audio, tempFile);
+                return tempFile;
+            } catch (JAXBException jaxbException) {
+                throw new IOException(jaxbException);
+            }
+        }
+
+        return audioFile;
+    }
+
     @Override
-    public void execute() {
+    public void execute() throws IOException {
```

Comment:
we don't need throws IOException as we are catching it and throwing a command exception.

Reply:
Done